### ISSUE TITLE
Several issues regarding type conversion

### ISSUE THREAD

**Issue by tujie-jiangye** [MeiGen-AI/InfiniteTalk#114]
I found many type conversions of amp.autocast(dtype=torch.float32) in the code and would like to ask the author about the reason for this approach. I found that it is not quite consistent with the Settings of wan2.1. During the training process, is the training based on the float32 type or bf16.

Thanks!

`

class WanAttentionBlock(nn.Module):

    def __init__(self,
                 cross_attn_type,
                 dim,
                 ffn_dim,
                 num_heads,
                 window_size=(-1, -1),
                 qk_norm=True,
                 cross_attn_norm=False,
                 eps=1e-6,
                 output_dim=768,
                 norm_input_visual=True,
                 class_range=24,
                 class_interval=4):
        ...
        

    def forward(
        self,
        x, 
        e,
        seq_lens, 
        grid_sizes,
        freqs,
        context,
        context_lens, 
        audio_embedding=None, 
        ref_target_masks=None, 
        human_num=None,
    ):

        dtype = x.dtype
        assert e.dtype == torch.float32
        with amp.autocast(dtype=torch.float32):
            e = (self.modulation.to(e.device) + e).chunk(6, dim=1)
        assert e[0].dtype == torch.float32

        # self-attention
        y, x_ref_attn_map = self.self_attn(
            (self.norm1(x).float() * (1 + e[1]) + e[0]).type_as(x), seq_lens, grid_sizes,
            freqs, ref_target_masks=ref_target_masks)
        with amp.autocast(dtype=torch.float32):
            x = x + y * e[2]
        
        x = x.to(dtype)

        # cross-attention of text
        x = x + self.cross_attn(self.norm3(x), context, context_lens)

        # cross attn of audio
        x_a = self.audio_cross_attn(self.norm_x(x), encoder_hidden_states=audio_embedding,
                                        shape=grid_sizes[0], x_ref_attn_map=x_ref_attn_map, human_num=human_num)
        x = x + x_a

        y = self.ffn((self.norm2(x).float() * (1 + e[4]) + e[3]).to(dtype))
        with amp.autocast(dtype=torch.float32):
            x = x + y * e[5]

        x = x.to(dtype)

        return x

`
